protocol/contracts/oud/OudRedeemer.sol

```diff
@@ -63,58 +66,79 @@ contract OudRedeemer is IOudRedeemer, Ownable {
   /// @dev Allows the tpi to be adjusted.
   /// @param value is the new TPI
   function setTreasuryPriceIndex(uint256 value) external override onlyOwner {
-    emit TreasuryPriceIndexSet (treasuryPriceIndex, value);
+    emit TreasuryPriceIndexSet(treasuryPriceIndex, value);
     treasuryPriceIndex = value;
   }
 
-/// @dev set address of stable coin used for minting Temple 
-/// @param newStable updated address fo 'stable', cannot be 0
+  /// @dev set address of stable coin used for minting Temple 
+  /// @param newStable updated address fo 'stable', cannot be 0
   function setStableCoin(address newStable) external override onlyOwner {
     if (newStable == address(0)) revert AddressZero();
-    emit StableCoinSet (address(stable), newStable);
+    emit StableCoinSet(address(stable), newStable);
     stable = IERC20(newStable);
   }
 
   /**
-   * @notice Provides a quote for 'oudAmount' with relevant token addresses:
-   *  'oudAmount' provided, 'stable' token required, 'templeAmount' that will be minted to caller
-   * @param oudAmount amount of Oud that will be redeemed
-   * @return _stableAmount amount of Stable token required at tpi
-   * @return _templeAmount amount of Temple that will be minted
-   */
-  function getQuote(
+    * @notice Provides a quote for how much stable is required from the user and 
+    * how much temple will be minted to the user, given an `oudAmount`
+    * @param oudAmount The amount of Oud that will be redeemed
+    * @return _stableAmount The amount of Stable token required from the user
+    * @return _templeAmount The amount of Temple that will be minted to the user
+    */
+  function redeemQuote(
     uint256 oudAmount
-  ) external view returns (uint256 _stableAmount, uint256 _templeAmount) {
-    return (_getStableAmount(oudAmount), oudAmount);
+  ) external override view returns (uint256 _stableAmount, uint256 _templeAmount) {
+    _stableAmount = _getStableAmount(oudAmount, treasuryPriceIndex);
+    _templeAmount = oudAmount;
   }
-/**
-  * @notice Redeems Oud and Stable token (@ tpi) for Temple
-  * @dev requires allowances for Oud and Stable Token
-  * @param oudAmount The amount of Oud to be redeemed for Temple
-  * @return templeAmount The amount of Temple that has been minted to sender.
-  */ 
-  function redeem(uint256 oudAmount) external returns (uint256 templeAmount) {
+
+  /**
+    * @notice Redeems Oud and Stable token (@ tpi) for Temple
+    * @dev requires allowances for Oud and Stable Token
+    * @param oudAmount The amount of Oud to be redeemed for Temple
+    * @return stableAmount The amount of Stable token required from the user
+    * @return templeAmount The amount of Temple that has been minted to sender.
+    */ 
+  function redeem(uint256 oudAmount) external override returns (uint256 stableAmount, uint256 templeAmount) {
     if (oudAmount == 0) revert RedeemAmountZero();
-    uint256 _stableAmount = _getStableAmount(oudAmount);
+    uint256 _tpi = treasuryPriceIndex;
```

Comment:
Reading from storage (sstore) is relatively expensive for gas. So if you read more than once, then it pays to put it in a locally scoped var on the stack.